### PR TITLE
Singlem update

### PR DESCRIPTION
**Pull request by wwood**
SingleM hasn't worked for a while, fixing now it has been released on conda recently (0.14.0).

To do:
- [x] Update singlem commands
- [x] Fix appraise plot bug - @AroneyS ?
- [x] Require SINGLEM_METAPACKAGE_PATH to be defined a la GTDBTK data env variable
- [x] Update integration tests to check singlem is working correctly there
- [x] Merge this into dev, not main

Current plotting bug:
```
$ SINGLEM_METAPACKAGE_PATH=/work/microbiome/db/singlem/S3.2.0.GTDB_r214.metapackage_20230428.smpkg.zb aviary recover -o out -1 test/data/wgsim.1.fq.gz -2 test/data/wgsim.2.fq.gz -n 32 -t 32
...

$ tail out/data/singlem_out/singlem_log.txt -n20
...
10/02/2023 08:39:07 AM INFO: Generating plot for marker: S3.52.rpsC_bact
[2023-10-01T22:39:07Z INFO  bird_tool_utils::clap_utils] Smafa version 0.6.1
[2023-10-01T22:39:07Z INFO  smafa::cluster] Clustering ..
[2023-10-01T22:39:07Z INFO  smafa::cluster] Clustering complete, took 0 seconds. Clustered 9 sequences into 2 clusters.
Traceback (most recent call last):
  File "/mnt/hpccs01/work/microbiome/conda/7b0823f7cd81a673d44f7e4adbc747aa_/bin/singlem", line 1021, in <module>
    app.plot(
  File "/mnt/hpccs01/work/microbiome/conda/7b0823f7cd81a673d44f7e4adbc747aa_/lib/python3.10/site-packages/singlem/appraisal_result.py", line 80, in plot
    self._plot_gene(
  File "/mnt/hpccs01/work/microbiome/conda/7b0823f7cd81a673d44f7e4adbc747aa_/lib/python3.10/site-packages/singlem/appraisal_result.py", line 269, in _plot_gene
    self._plot_scale(axis, max_total_count)
  File "/mnt/hpccs01/work/microbiome/conda/7b0823f7cd81a673d44f7e4adbc747aa_/lib/python3.10/site-packages/singlem/appraisal_result.py", line 352, in _plot_scale
    overlap = (ylim[1]-ylim[0]-sum(sides))/(len(sides)-1)
ZeroDivisionError: float division by zero
```

I think there is 2 things here - it shouldn't error, and should generate the plot. Also, it shouldn't choose a domain-specific gene to plot, it should use one that targets both Archaea and Bacteria.

### CODE REVIEW

File: aviary/modules/binning/binning.smk [hunk 1]
```diff
@@ -769,13 +769,14 @@ rule singlem_appraise:
...
-        "singlem pipe --threads {threads} --sequences {params.fasta} --otu_table data/singlem_out/assembly.otu_table.csv >> {log} 2>&1; "
-        "singlem appraise --metagenome_otu_tables {input.metagenome} --genome_otu_tables data/singlem_out/genomes.otu_table.csv "
-        "--assembly_otu_table data/singlem_out/assembly.otu_table.csv "
-        "--plot data/singlem_out/singlem_appraise.svg --output_binned_otu_table data/singlem_out/binned.otu_table.csv "
-        "--output_unbinned_otu_table data/singlem_out/unbinned.otu_table.csv > data/singlem_out/singlem_appraisal.tsv 2>> {log} || "
-        "echo 'SingleM Errored, please check data/singlem_out/singlem_log.txt' && touch data/singlem_out/singlem_appraisal.tsv"
+        # We use bash -c so that a non-zero exitstatus of the non-final commands doesn't cause the rule (and therefore aviary) to fail
+        "bash -c 'singlem pipe --threads {threads} --genome-fasta-file bins/final_bins/*.fna --otu-table data/singlem_out/genomes.otu_table.csv && "
+        "singlem pipe --threads {threads} --genome-fasta-file {params.fasta} --otu-table data/singlem_out/assembly.otu_table.csv && "
```
Comment (AroneyS) [inline aviary/modules/binning/binning.smk:774]: Why did you remove the intermediate logging?

Comment (wwood) [reply aviary/modules/binning/binning.smk:774]: It is all within the `bash -c` so all of it is logged, right?

Comment (AroneyS) [reply aviary/modules/binning/binning.smk:774]: Only appraise was logged when I ran it in test_integration

Comment (AroneyS) [reply aviary/modules/binning/binning.smk:774]: Does the logging part need to be outside the quotes?

Comment (wwood) [reply aviary/modules/binning/binning.smk:774]: Ah sorry the single quote should be before the `2> {log}` not after - can you change/test please?

**Comment by AroneyS** [comment]
Plot bug fixed here: https://github.com/wwood/singlem/pull/145
Needs to be pulled and new SingleM version?

**Comment by AroneyS** [comment]
@wwood Just need a new version of SingleM and to check that integration tests pass.

**Review by AroneyS — APPROVED** [review]
